internal/tss/router/router.go

```diff
@@ -0,0 +1,89 @@
+package router
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/stellar/go/xdr"
+	"github.com/stellar/wallet-backend/internal/entities"
+	"github.com/stellar/wallet-backend/internal/tss"
+)
+
+type Router interface {
+	Route(payload tss.Payload) error
+}
+
+type RouterConfigs struct {
+	RPCCallerChannel      tss.Channel
+	ErrorJitterChannel    tss.Channel
+	ErrorNonJitterChannel tss.Channel
+	WebhookChannel        tss.Channel
+}
+
+type router struct {
+	RPCCallerChannel      tss.Channel
+	ErrorJitterChannel    tss.Channel
+	ErrorNonJitterChannel tss.Channel
+	WebhookChannel        tss.Channel
+}
+
+var _ Router = (*router)(nil)
+
+var FinalErrorCodes = []xdr.TransactionResultCode{
+	xdr.TransactionResultCodeTxSuccess,
+	xdr.TransactionResultCodeTxFailed,
+	xdr.TransactionResultCodeTxMissingOperation,
+	xdr.TransactionResultCodeTxInsufficientBalance,
+	xdr.TransactionResultCodeTxBadAuthExtra,
+	xdr.TransactionResultCodeTxMalformed,
+}
+
+var NonJitterErrorCodes = []xdr.TransactionResultCode{
+	xdr.TransactionResultCodeTxTooEarly,
+	xdr.TransactionResultCodeTxTooLate,
+	xdr.TransactionResultCodeTxBadSeq,
+}
+
+var JitterErrorCodes = []xdr.TransactionResultCode{
+	xdr.TransactionResultCodeTxInsufficientFee,
+	xdr.TransactionResultCodeTxInternalError,
+}
+
+func NewRouter(cfg RouterConfigs) Router {
+	return &router{
+		RPCCallerChannel:      cfg.RPCCallerChannel,
+		ErrorJitterChannel:    cfg.ErrorJitterChannel,
+		ErrorNonJitterChannel: cfg.ErrorNonJitterChannel,
+		WebhookChannel:        cfg.WebhookChannel,
+	}
+}
+
+func (r *router) Route(payload tss.Payload) error {
+	var channel tss.Channel
+	if payload.RpcSubmitTxResponse.Status.Status() != "" {
+		switch payload.RpcSubmitTxResponse.Status.Status() {
+		case string(tss.NewStatus):
+			channel = r.RPCCallerChannel
+		case string(entities.TryAgainLaterStatus):
+			channel = r.ErrorJitterChannel
+		case string(entities.ErrorStatus):
+			if payload.RpcSubmitTxResponse.Code.OtherCodes == tss.NoCode {
+				if slices.Contains(JitterErrorCodes, payload.RpcSubmitTxResponse.Code.TxResultCode) {
+					channel = r.ErrorJitterChannel
+				} else if slices.Contains(NonJitterErrorCodes, payload.RpcSubmitTxResponse.Code.TxResultCode) {
+					channel = r.ErrorNonJitterChannel
+				} else if slices.Contains(FinalErrorCodes, payload.RpcSubmitTxResponse.Code.TxResultCode) {
+					channel = r.WebhookChannel
+				}
+			}
+		default:
+			// Do nothing for PENDING / DUPLICATE statuses
+			return nil
+		}
+	}
+	if channel == nil {
+		return fmt.Errorf("payload could not be routed - channel is nil")
+	}
+	channel.Send(payload)
+	return nil
+}
```

Comment:
💡 No problem with the logic, just an idea for improved readability:

```suggestion
func (r *router) Route(payload tss.Payload) error {
	switch payload.RpcSubmitTxResponse.Status.Status() {
	case string(tss.NewStatus):
		r.RPCCallerChannel.Send(payload)

	case string(entities.TryAgainLaterStatus):
		r.ErrorJitterChannel.Send(payload)

	case string(entities.ErrorStatus):
		if payload.RpcSubmitTxResponse.Code.OtherCodes == tss.NoCode {
			if slices.Contains(JitterErrorCodes, payload.RpcSubmitTxResponse.Code.TxResultCode) {
				r.ErrorJitterChannel.Send(payload)
			} else if slices.Contains(NonJitterErrorCodes, payload.RpcSubmitTxResponse.Code.TxResultCode) {
				r.ErrorNonJitterChannel.Send(payload)
			} else if slices.Contains(FinalErrorCodes, payload.RpcSubmitTxResponse.Code.TxResultCode) {
				r.WebhookChannel.Send(payload)
			} else {
				return fmt.Errorf("payload could not be routed, status %+v", payload.RpcGetIngestTxResponse.Status)
			}
		}

	// Do nothing for PENDING / DUPLICATE statuses
	case string(entities.PendingStatus), string(entities.DuplicateStatus):
		return nil

	default:
		return fmt.Errorf("unexpected status %+v", payload.RpcGetIngestTxResponse.Status)
	}

	return nil
}
```

Reply:
That is more readable, but the thing is, I want to be able to pass nil values for some/all channels. Because with the ingestor for example, it will only ever need the webhook channel. And if im passing nil values for these, I think I should explicitly check to see if the channels are nil before sending payloads to them